Treppenhauslichtsteuerung/module.php

```diff
@@ -167,43 +250,59 @@ public function UpdateRemaining()
+            {
+                $inputTriggers = json_decode($this->ReadPropertyString('InputTriggers'), true);
+                foreach ($inputTriggers as $variable) {
+                    if ($variable['VariableID'] == $outputID) {
+                        return true;
+                    }
+                }
+                return false;
+            };
 
-        if (IPS_GetVariable($outputID)['VariableType'] == VARIABLETYPE_BOOLEAN) {
-            RequestAction($outputID, $Value);
-        } else {
-            $profileName = $this->GetProfileName($variable);
-            //Quit if output variable has no profile
-            if (!IPS_VariableProfileExists($profileName)) {
-                echo $this->Translate('The output variable of the Treppenhauslichtsteuerung has no variable profile. Please choose a variable with a variable profile or add a variable profile to the output variable.');
-                return;
+            if ($Value && $isTrigger($outputID) && GetValue($outputID)) {
+                continue;
             }
-            $maxValue = IPS_GetVariableProfile($profileName)['MaxValue'];
-            $minValue = IPS_GetVariableProfile($profileName)['MinValue'];
+            $outputVariable = IPS_GetVariable($outputID);
+            if ($outputVariable['VariableType'] == VARIABLETYPE_BOOLEAN) {
+                $actionValue = $Value;
+                if ($this->profileReversed($outputID)) {
+                    $actionValue = !$Value;
+                }
+                RequestAction($outputID, $actionValue);
+            } else {
+                //If we are enabling analog devices we want to switch to the maximum value (e.g. 100%)
+                $profile = IPS_GetVariableProfile($this->GetProfileName($outputVariable));
 
-            //Quit if min is greater than max value
-            if ($maxValue - $minValue <= 0) {
-                echo $this->Translate('The profile of the output variable has no defined max value. Please update the max value or choose another profile.');
-                return;
-            }
+                $maxValue = $profile['MaxValue'];
+                $minValue = $profile['MinValue'];
 
-            //If we are enabling analog devices we want to switch to the maximum value (e.g. 100%)
-            $actionValue = 0;
-            if ($Value) {
-                $actionValue = $maxValue;
-            } else {
-                $actionValue = $minValue;
-            }
+                $actionValue = 0;
+                if ($this->profileReversed($outputID)) {
+                    if ($Value) {
+                        $actionValue = $minValue;
+                    } else {
+                        $actionValue = $maxValue;
+                    }
+                } else {
+                    if ($Value) {
+                        $actionValue = $maxValue;
+                    } else {
+                        $actionValue = $minValue;
+                    }
+                }
```

Comment:
Das kann man in einem if else zusammenbasteln, alternativ auch per "? :" Notation